scripts/run_lighthouse_tests.py

```diff
@@ -49,12 +60,125 @@ def run_lighthouse_checks():
     for line in iter(process.stdout.readline, ''):
         python_utils.PRINT(line[:-1])
 
+    if process.returncode != 0:
+        python_utils.PRINT('Checks failed view details above')
+        sys.exit(1)
 
-def main():
-    """Runs lighthouse checks and deletes reports."""
-    setup_and_install_dependencies()
+
+def start_google_app_engine_server(port=APP_ENGINE_PORT):
+    """Start the Google App Engine server."""
+    app_yaml_filepath = 'app.yaml'
+
+    python_utils.PRINT('Starting oppia server...')
+
+    server_process = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' % (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_HOME,
+            port, app_yaml_filepath), shell=True)
+    RUNNING_PROCESSES.append(server_process)
```

Comment:
Since we can specify the ports in lighthouse.yml, could we run the server through `scritps.start` and then just connect nginx to it?